ocrd_calamari/recognize.py

```diff
@@ -36,13 +36,28 @@
 from calamari_ocr.ocr import MultiPredictor
 from calamari_ocr.ocr.voting import voter_from_proto
 from calamari_ocr.proto import VoterParams
+from tensorflow import config as tensorflow_config
 
 # ruff: isort: on
 
 from ocrd_calamari.config import OCRD_TOOL
 
 TOOL = "ocrd-calamari-recognize"
 
+BATCH_SIZE = 64
+def batched(iterable, n):
+    # batched('ABCDEFG', 3) → ABC DEF G
+    if n < 1:
+        raise ValueError('n must be at least one')
+    iterator = iter(iterable)
+    while batch := tuple(itertools.islice(iterator, n)):
+        # implement poor man's batch bucketing to avoid OOM:
+        maxlen = max(image.shape[1] for image in batch)
+        if maxlen * n > 32000 and n > 1:
+            yield from batched(batch, n//2)
+        else:
+            yield batch
+itertools.batched = batched
```

Comment:
Now this always overrides itertools.batched, should we not just use batched directly?

(Duplicate comment, I had commented in your repo at first)

Reply:
I know, it's hacky, but I figured it would be the best place to solve both problems:
- Py38 itertools not having `batched` yet
- C1 not providing batch bucketing, so vanilla `itertools.batched` would still be insufficient

Reply:
Do you want me to separate these concerns by defining our own `batched_length_limited` and delegating to `itertools.batched` or a reimplementation thereof?

Reply:
Defining our own `batched_length_limited` sounds good!

I'm mainly opposed against overwriting `itertools.batched` if it isn't just a backport with ~100% compatibility.

Reply:
So be it: https://github.com/bertsky/ocrd_calamari/commit/842bd9287ed0673a76f4cb22a98675236adb9909